src/asRendition.tsx

```diff
@@ -92,11 +92,16 @@ export const withStyledSystem = (child: React.ComponentType) => {
 
 export const withTooltip = (Base: React.ComponentType) => {
 	return React.forwardRef(({ ...props }: any, ref: any) => {
-		if (props.tooltip) {
-			props = tooltip.bindProps(props);
+		if (!props.tooltip) {
+			return <Base {...props} ref={ref} />;
 		}
+		const tooltipProps = tooltip.bindProps(props);
 		delete props.tooltip;
-		return <Base {...props} ref={ref} />;
+		return (
+			<div {...tooltipProps}>
```

Comment:
should we maybe use `span` instead? This might cause some layouting issues if the base was originally inline

Reply:
That could still have a bad impact if we are using flexbox on the parent :thinking:

Reply:
Nice catch @sradevski, I'm thinking that span can be also a problem for components like `Flex` or `Box` since this would break "children's logic". I therefore also added a last check on the disabled prop.